apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/MutationsTest.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    public class MutationsTest
+    {
+        public static IEnumerable<object[]> ValidFamilyNames { get; } = new List<object[]>
+        {
+            new [] { "a" },
+            new [] { "A" },
+            new [] { "-_.aZ09" }
+        };
+
+        public static IEnumerable<object[]> InvalidFamilyNames { get; } = new List<object[]>
+        {
+            new [] { default(string) },
+            new [] { "" },
+            new [] { "abc*" },
+            new [] { "abc " },
+            new [] { "a=b" }
+        };
+
+        [Theory]
+        [MemberData(nameof(ValidFamilyNames))]
+        public void DeleteFromColumn(string familyName)
+        {
+            var mutation =
+                Mutations.DeleteFromColumn(familyName, "cq1", new BigtableVersionRange(1, 2));
+            Assert.NotNull(mutation.DeleteFromColumn);
+            Assert.Equal(familyName, mutation.DeleteFromColumn.FamilyName);
+            Assert.Equal("cq1", mutation.DeleteFromColumn.ColumnQualifier.ToStringUtf8());
+            Assert.Equal(1, mutation.DeleteFromColumn.TimeRange.StartTimestampMicros);
```

Comment:
I thought micros always had to be a multiple of 1000, as per another email thread? Maybe I'm confused.

Reply:
I just realized that as I was writing these integration tests and then started that thread (was there another one I forgot about)? Yes, I need to fix these tests and the `BigtableVersion` class as well so the multiple-of-1000s requirement is hidden for people wanting to use a custom versioning scheme. I was thinking of having the `BigtableVersion(long)` constructor interpret the param as millis, not micros, and having `BigtableVersion.Value` return millis, but still store micros internally. I'm a little worried about new granularities getting added in the future though and that this locks us into that granularity. But perhaps in the future if that were to ever happen these could be the defaults and we could add a constructor with an extra parameter for the granularity and a `GetValue(...)` method taking a granularity. I was going to do this all in another PR though. Thoughts?

Reply:
Eek - yes, it's awkward. Best to be as granularity-neutral as possible, I think. I suggest we tackle this in a separate PR, where the effects will be clearer.